encoding/ssz/query/ssz_interface.go

```diff
@@ -0,0 +1,69 @@
+package query
+
+import (
+	"errors"
+)
+
+type SSZIface interface {
+	HashTreeRoot() ([32]byte, error)
+	MarshalSSZ() ([]byte, error)
+	UnmarshalSSZ([]byte) error
+	// HashTreeRootWith(hh *ssz.Hasher) error
+}
+
+// HashTreeRoot calls the HashTreeRoot method on the stored interface if it implements SSZIface.
+// Returns the 32-byte hash tree root or an error if the interface doesn't support hashing.
+func (info *sszInfo) HashTreeRoot() ([32]byte, error) {
+	if info == nil {
+		return [32]byte{}, errors.New("sszInfo is nil")
+	}
+
+	if info.iface == nil {
+		return [32]byte{}, errors.New("not stored iface")
```

Comment:
```suggestion
		return [32]byte{}, errors.New("sszInfo.iface is nil")
```
I looked at the existing code and there's already one error message about a nil field, so maybe let's keep it unified.
```
if info.containerInfo == nil {
		return nil, errors.New("sszInfo.containerInfo is nil")
	}
```

Reply:
Also agreed. It is better to keep it homogeneous